### ISSUE TITLE
RandomizedPCA is deprecated

### ISSUE THREAD

**Issue by drczuckerman** [udacity/ud120-projects#154]
I'm currently using scikit-learn-0.19, and I'm getting this warning that RandomizedPCA is deprecated:

`C:\ProgramData\Anaconda2\lib\site-packages\sklearn\utils\deprecation.py:58: DeprecationWarning: Class RandomizedPCA is deprecated; RandomizedPCA was deprecated in 0.18 and will be removed in 0.20. Use PCA(svd_solver='randomized') instead. The new implementation DOES NOT store whiten ``components_``. Apply transform to get them.`

I've tried to use PCA in pca/eigenfaces.py:

`pca = PCA(n_components=n_components, whiten=True, svd_solver="randomized").fit(X_train)`

I've not found any example that indicates how I'm supposed to use "transform" as the warning message indicates.

Here are my results using RandomizedPCA:
```                   precision    recall  f1-score   support

     Ariel Sharon       0.70      0.54      0.61        13
     Colin Powell       0.81      0.87      0.84        60
  Donald Rumsfeld       0.94      0.63      0.76        27
    George W Bush       0.83      0.98      0.90       146
Gerhard Schroeder       0.95      0.76      0.84        25
      Hugo Chavez       1.00      0.47      0.64        15
       Tony Blair       0.94      0.81      0.87        36

      avg / total       0.86      0.85      0.84       322

[[  7   2   0   4   0   0   0]
 [  1  52   0   7   0   0   0]
 [  1   2  17   6   0   0   1]
 [  0   3   0 143   0   0   0]
 [  0   1   0   4  19   0   1]
 [  0   3   0   4   1   7   0]
 [  1   1   1   4   0   0  29]]
```
Here are my results trying to use PCA:
```
                   precision    recall  f1-score   support

     Ariel Sharon       0.56      0.69      0.62        13
     Colin Powell       0.74      0.87      0.80        60
  Donald Rumsfeld       0.76      0.81      0.79        27
    George W Bush       0.93      0.87      0.90       146
Gerhard Schroeder       0.76      0.76      0.76        25
      Hugo Chavez       0.73      0.53      0.62        15
       Tony Blair       0.88      0.83      0.86        36

      avg / total       0.84      0.83      0.83       322

[[  9   0   3   1   0   0   0]
 [  2  52   1   4   0   1   0]
 [  4   0  22   1   0   0   0]
 [  1  11   2 127   3   1   1]
 [  0   2   0   1  19   1   2]
 [  0   3   0   1   2   8   1]
 [  0   2   1   2   1   0  30]]
```
As you can see, I'm getting different, and in a lot of cases, worse results.

What do I need to do to get the same results as RandomizedPCA using PCA?

**Comment by dphuongnguyen** [comment]
I think the new way you use PCA in pca/eigenfaces.py is the right way. I am not sure what the warning meant by saying "transform" here.

About the result, I think it depends on the random_state when you split the train set and test set. 

For some reason, with the same random_state, it does not give the same result on the deprecated version and on the new version. Maybe for some other seed (different seeds for each version), it will yield the same result.

**Comment by Aaraeus** [comment]
This is still an issue in 0.20.2, and it also looks like the PIL library for imaging has been removed. Some googling recommends Pillow, but it just seems like the entire exercise is probably outdated. Hoping someone more skilled than I might have recoded the eignfaces.py code. I'll keep digging.

**Comment by venkat-amballa** [comment]
There is an example in the scikit-learn Official website. visit this [_link_ ](https://scikit-learn.org/stable/auto_examples/applications/plot_face_recognition.html#sphx-glr-auto-examples-applications-plot-face-recognition-py )for the **updated version of the code for pca/eigenfaces.py.**
